evalml/pipelines/component_graph.py

```diff
@@ -61,13 +61,35 @@ def _validate_component_dict(self):
                 component_input.endswith(".x") or component_input == "X"
                 for component_input in component_inputs
             )
-            has_target_input = any(
+            has_one_target_input = sum(
                 component_input.endswith(".y") or component_input == "y"
                 for component_input in component_inputs
             )
-            if not (has_feature_input and has_target_input):
+            if not has_feature_input:
```

Comment:
Kind of redundant / can be covered by case below. Kept for the specific ValueError, but open to just removing and having it encapsulated below 😂